### ISSUE TITLE
Multiple XmlElementAttribute attributes are not supported

### ISSUE THREAD

**Issue by charlie430** [xorets/ObjectXPathNavigator#4]
It seems ObjectXPathNavigator doesn't support multiple XmlElementAttribute attributes.

http://msdn.microsoft.com/en-us/library/system.xml.serialization.xmlelementattribute.aspx

``` C#
[XmlElement(typeof(int), ElementName = "ObjectNumber")]
[XmlElement(typeof(string), ElementName = "ObjectString")]
public ArrayList ExtraInfo;
```

After looking at sdf.XPath.NodeInfo, it seems for performance reasons, it caches the node name and namespace, however, in this case, there are multiple node names and namespaces that may be valid.

**Comment by xorets** [comment]
Hmm, yes, this is not supported for now. I really don't recall how ArrayList is currently serialized if it has XmlElement attribute on it. Is it goes with root element or w/o it. Initially I were not aiming to fully mimic XmlSerializer behavior, then it was requested and partially implemented. To the an extent suitable for task at hand. :)

As I see it is possible to implement, though. What could be done:
1. Make list of XmlElementAttributes in NodeInfo instead of one attribute.
2. Make ArrayList node policy that will look at Node.MemberInfo and emit names for its child elements according to types.

**Comment by charlie430** [comment]
Indeed, my local attempt to resolve the issue included step #1 you suggested. The following scenarios are also possible. There's already an EnumerableNodePolicy, so I don't think we need a new policy, just a slight alternation to the ones that already exist.

``` c#
[XmlElement("A", typeof(ElementA)]
[XmlElement("B", typeof(ElementB)]
public ObservableCollection<ElementBase> Elements { get; set; }

[XmlElement("A", typeof(ElementA)]
[XmlElement("B", typeof(ElementB)]
public ElementBase Element { get; set; }

[XmlElement("A", typeof(ElementA)]
[XmlElement("B", typeof(ElementB)]
public object Element { get; set; }
```

**Comment by xorets** [comment]
Well, EnumerableNodePolicy is good place, but I believe it is used not everywhere. There are special policies for more specific types. Bad side of enumerable one is that it must copy all of its content to other indexable collection. If collection already could be accessed by index, then this step is not needed.
